Newer CBC no longer works with Pyomo

In testing CBC 2.10.3 (from https://ampl.com/dl/open/cbc/cbc-linux64.zip) on RHEL7 with Python 3.7, these errors/failures are now occurring: 
```
======================================================================
ERROR: test_cbc_nl_symbolic_labels (pyomo.solvers.tests.checks.test_writers.transplant_class.<locals>.C)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mmundt/devel-pyomo/pyomo/pyomo/solvers/tests/checks/test_writers.py", line 72, in writer_test
    load_solutions)
  File "/home/mmundt/devel-pyomo/pyomo/pyomo/solvers/tests/models/base.py", line 126, in solve
    **io_options)
  File "/home/mmundt/devel-pyomo/pyomo/pyomo/opt/base/solvers.py", line 600, in solve
    "Solver (%s) did not exit normally" % self.name)
pyutilib.common._exceptions.ApplicationError: Solver (cbc) did not exit normally
-------------------- >> begin captured logging << --------------------
pyomo.opt: ERROR: Solver (cbc) returned non-zero return code (-11)
pyomo.opt: ERROR: Solver log:
CBC 2.10.3: printingOptions=all
 
--------------------- >> end captured logging << ---------------------
 
======================================================================
FAIL: test_optimal_mip (pyomo.solvers.tests.checks.test_CBCplugin.TestCBC)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mmundt/devel-pyomo/pyomo/pyomo/solvers/tests/checks/test_CBCplugin.py", line 121, in test_optimal_mip
    self.assertEqual(1.0, results.problem.lower_bound)
AssertionError: 1.0 != -1.0
 
======================================================================
FAIL: test_unbounded_mip (pyomo.solvers.tests.checks.test_CBCplugin.TestCBC)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mmundt/devel-pyomo/pyomo/pyomo/solvers/tests/checks/test_CBCplugin.py", line 106, in test_unbounded_mip
    self.assertEqual(TerminationCondition.unbounded, results.solver.termination_condition)
AssertionError: EnumValue(<pyutilib.enum.enum.Enum object at 0x7fb23c9a9e10>, 11, 'unbounded') != EnumValue(<pyutilib.enum.enum.Enum object at 0x7fb23c9a9e10>, 12, 'infeasible')
```

These need to be addressed before introducing CBC testing in the Windows Github Actions (#1292).

--- a/pyomo/solvers/tests/checks/test_no_solution_behavior.py
+++ b/pyomo/solvers/tests/checks/test_no_solution_behavior.py
@@ -75,15 +75,6 @@ def create_test_method(model,
             # file with garbage values in it for a failed solve
             self.assertEqual(len(results.solution), 1)
 
-    # 03/23/2021: IDAES-ext added CBC 2.10.4 to their official release
-    #             This is causing failures in this test.
-    #             Manually turning off CBC tests until a solution can be found.
-    #             - mrmundt
-    if solver == 'cbc':
-        def skipping_test(self):
-            self.skipTest('SKIP: cbc currently does not work.')
-        return skipping_test
-
     # Skip this test if the status is 'skip'
     if test_case.status == 'skip':
         def skipping_test(self):

--- a/pyomo/solvers/tests/checks/test_pickle.py
+++ b/pyomo/solvers/tests/checks/test_pickle.py
@@ -103,15 +103,6 @@ def create_test_method(model, solver, io,
         # then unpickle and load status
         inst, res = pickle.loads(pickle.dumps([instance3,status3]))
 
-    # 03/23/2021: IDAES-ext added CBC 2.10.4 to their official release
-    #             This is causing failures in this test.
-    #             Manually turning off CBC tests until a solution can be found.
-    #             - mrmundt
-    if solver == 'cbc':
-        def skipping_test(self):
-            self.skipTest('SKIP: cbc currently does not work.')
-        return skipping_test
-
     # Skip this test if the status is 'skip'
     if test_case.status == 'skip':
         def skipping_test(self):

--- a/pyomo/solvers/tests/checks/test_writers.py
+++ b/pyomo/solvers/tests/checks/test_writers.py
@@ -118,15 +118,6 @@ def create_test_method(model,
         except OSError:
             pass
 
-    # 03/23/2021: IDAES-ext added CBC 2.10.4 to their official release
-    #             This is causing failures in this test.
-    #             Manually turning off CBC tests until a solution can be found.
-    #             - mrmundt
-    if solver == 'cbc':
-        def skipping_test(self):
-            self.skipTest('SKIP: cbc currently does not work.')
-        return skipping_test
-
     # Skip this test if the status is 'skip'
     if test_case.status == 'skip':
         def skipping_test(self):

--- a/pyomo/solvers/tests/testcases.py
+++ b/pyomo/solvers/tests/testcases.py
@@ -106,8 +106,25 @@ ExpectedFailures['glpk', 'mps', 'LP_duals_maximize'] = \
 
 ExpectedFailures['cbc', 'nl', 'MILP_unbounded'] = \
     (lambda v: v <= _trunk_version,
-     "Cbc fails to report a MILP model as unbounded when it"
-     "is defined as an NL file.")
+     "Cbc fails to report an unbounded MILP model as unbounded through "
+     "the NL interface (through 2.9.x), and fails with invalid free() "
+     "(in 2.10.x).")
+
+ExpectedFailures['cbc', 'nl', 'LP_unbounded'] = \
+    (lambda v: v[:2] == (2, 10),
+     "Cbc fails (invalid free()) for unbounded LP models through "
+     "the NL interface in 2.10.x versions "
+     "(reported upstream as coin-or/Cbc#389)")
+
+ExpectedFailures['cbc', 'nl', 'SOS1_simple'] = \
+    (lambda v: v[:2] == (2, 10),
+     "Cbc segfaults for SOS constraints in the NL interface "
+     "(reported upstream as coin-or/Cbc#388)")
+
+ExpectedFailures['cbc', 'nl', 'SOS2_simple'] = \
+    (lambda v: v[:2] == (2, 10),
+     "Cbc segfaults for SOS constraints in the NL interface "
+     "(reported upstream as coin-or/Cbc#388)")
 
 #
 # PICO

